tests/contracts/wallet_2.scilla

```diff
@@ -0,0 +1,392 @@
+(* This wallet does not allow adding or removing   *)
+(* owners, or changing the number of required      *)
+(* signatures. To do any of those things, perform  *)
+(* the following steps:                            *)
+(*                                                 *)
+(* 1. Deploy a new wallet with owners and  *)
+(*    required_signatures set to the new values.   *)
+(*    MAKE SURE THAT THE NEW WALLET HAS BEEN       *)
+(*    SUCCESFULLY DEPLOYED WITH THE CORRECT        *)
+(*    PARAMETERS BEFORE CONTINUING!                *)
+(* 2. Invoke the SubmitTransaction transition on   *)
+(*    the old wallet with the following            *)
+(*    parameters:                                  *)
+(*    recipient : The address of the new wallet    *)
+(*    amount : The _balance of the old wallet      *)
+(*    tag : "AddFunds"                             *)
+(* 3. Have (a sufficient number of) the owners of  *)
+(*    the old contract invoke the SignTransaction  *)
+(*    transition on the old wallet. The parameter  *)
+(*    transactionId should be set to the Id of the *)
+(*    transaction created in step 2.               *)
+(* 4. Have one of the owners of the old contract   *)
+(*    invoke the ExecuteTransaction transition on  *)
+(*    the old contract. This will cause the entire *)
+(*    balance of the old contract to be            *)
+(*    transferred to the new wallet. Note that no  *)
+(*    un-executed transactions will be transferred *)
+(*    to the new wallet along with the funds.      *)
+(*                                                 *)
+(* WARNING: If a sufficient number of owners lose  *)
+(* their private keys, or for any other reason are *)
+(* unable or unwilling to sign for new             *)
+(* transactions, the funds in the wallet will be   *)
+(* locked forever. It is therefore a good idea to  *)
+(* set required_signatures to a value strictly     *)
+(* less than the number of owners, so that the     *)
+(* remaining owners can retrieve the funds should  *)
+(* such a scenario occur.                          *)
+(*                                                 *)
+(* If an owner loses his private key, the          *)
+(* remaining owners should move the funds to a new *)
+(* wallet (using the workflow described above) to  *)
+(* ensure that funds are not locked if another     *)
+(* owner loses his private key. The owner who      *)
+(* originally lost his private key can generate a  *)
+(* new key, and the corresponding address be added *)
+(* to the new wallet, so that the same set of      *)
+(* persons own the new wallet.                     *)
+(*                                                 *)
+(***************************************************)
+contract Wallet
+(
+owners_list         : List ByStr20,
+required_signatures : Uint32
+)
+
+(* Funds are not allowed to be added if the contract is not valid *)
+field validity_checked : Bool = False
+field contract_valid   : Bool = False
```

Comment:
A possible alternative to having two boolean variables here is to have a custom data type with 3 constructors reflecting the possible states of the contract validity. I personally would find this approach a bit more readable, but you mileage may vary, of course.

Reply:
Fixed.